src/cargo/util/context/mod.rs

```diff
@@ -973,8 +974,8 @@ impl GlobalContext {
             }
         } else {
             output.extend(
-                env_val
-                    .split_whitespace()
+                Shlex::new(env_val)
+                    .into_iter()
                     .map(|s| (s.to_string(), def.clone())),
```

Comment:
We'll need a test for this.

Ideally, the PR gets split up into two commits
- The first adds the test, showing the bad behavior
- This commit updates the test showing the good behavior

As discussed at https://doc.crates.io/contrib/process/working-on-cargo.html#submitting-a-pull-request

Reply:
Noted! Rewriting git history and using this occasion to add a test.